GiEnJul/ClientApp/src/components/admin/AdminPage.tsx

```diff
@@ -21,15 +24,45 @@ function AdminPage() {
     });
   }, []);
 
-  if (userAccessToken !== "") {
+  if (location === null) {
     return (
-      <>
-        <NavMenuAdmin />
-        <LogOutButton />
-        <AdminTab accessToken={userAccessToken} />
-      </>
+      <ErrorPage
+        ErrorText={
+          "Du har ikke blitt tildelt en lokasjon enda, tilkall din GiEnJul admin"
+        }
+      />
     );
+  } else if (role !== undefined && location !== undefined) {
```

Comment:
This `else if` should probably switch place with the `else` on line 63? This if sentence can grow larger if more conditional checks is added. I think that the checks should be ran in if/else if conditions first and if the code doesn't enter any of them, the wanted page will be rendered.

Could it be an idea to change the logic for which page should be rendered into another function, and use a switch case here instead?

Reply:
It is not that much code, and not sure it's better to extract to another file. 
what do you mean by switching places with 35 and 63, then the else with <LoadingPage/> would always happen.
how would you like it?
if
else
  if
  if
I don't get it

Reply:
I think it would be a bit cleaner if we made a function e.g. `getPage`:
```typescript
const getPage = () => {
  if (location === null) return Page.NoLocation;
  else if (role === undefined || location === undefined) return Page.Loading;
  else if (role === "Admin") return Page.Admin;
  ...
}
```
The rendering could be changed to this then:
```typescript
switch (getPage()) {
  case Page.NoLocation:
    return <ErrorPage .../>;
  case Page.Loading:
    return <LoadingPage/>;
  case ...
}
```

Reply:
That would be very clean, I'll make that helper function. It has to be in the same file to have access to the variables tho

Reply:
Yeah that would be nice 😄

Reply:
```typescript 
function getPage() {
    if (location === null) return "NoLocation";
    else if (role === undefined || location === undefined) return "Loading";
    else if (role === "Unspecified") return "Unspecified";
    else if (role === "Admin") return "Admin";
    else if (role === "Institution") return "Institution";
  }
```
....
```typescript
switch (getPage()) {
...
case "Admin":
      return (
        <>
          <NavMenuAdmin />
          <AdminTab accessToken={userAccessToken} location={location} />
        </>
      );
```
The ts compiler kills me since it says that the string could be undefined

Reply:
The location variable can't be undefined since that would return the loading page. So added a {/*// @ts-ignore */} above the component and pushed it with f7aa4bf
have a look at it @joffe97

Reply:
Yeah looks like a flaw with typescript, because location will never be undefined here. You could probably change it into this: 
```typescript
<AdminTab accessToken={userAccessToken} location={location ?? ""} />
```